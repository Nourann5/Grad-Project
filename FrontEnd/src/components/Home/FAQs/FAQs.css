.faq .accordion-button::after{
    display: none;
}
.faq .accordion-button{
    background-color: transparent!important;
    padding: 35px 45px;
    box-shadow: none!important;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;
}
.faq .accordion-button.collapsed .faq__accordion-icon--open{
    transform: rotate(90deg)!important ;
}