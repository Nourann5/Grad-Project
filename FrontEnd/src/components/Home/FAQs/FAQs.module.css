
.faq{
    background-color: white;
    padding: 80px 0 100px;
}
.faq__heading-cont{
    position: relative;
    display: flex;
    align-items: center;
}
.faq__heading{
    font-size:40px;
    color: black;
    font-family:'PlayFairBold';
    text-align: center; 
    margin: 0 auto 30px;
}
.faq__icon {
    transform: translate(-150px, -60px);
    position: absolute;
}
.faq__accordion-item{
    margin-bottom: 25px;
    border-radius: 15px;
    box-shadow: none;
    border: 2px solid var(--main-color)!important;
    border-radius: 15px!important;
}
.faq__accordion-button{
    background-color: transparent;
    padding: 35px 45px;
    box-shadow: none;

}
.faq__accordion-header-content{
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    font-size: 20px;
    color: var(--text-color-dark);
    font-family:'PoppinsSemiBold';
}
.faq__accordion-body{
    padding: 0px 10px;
    font-size: 15px;
    color: var(--text-color-dark);
    font-family: 'PoppinsRegular';
    line-height: 24px;
    margin-top: 10px;
}
.faq__accordion-icon{
    position:absolute;
    transition: var(--transition)
}
.faq__accordion-icon:last-of-type{
    transform: rotate(0deg);
}
/* .faq__accordion-icon-cont.open .faq__accordion-icon--open{
    transform: rotate(90deg);
} */


.accordion .card{
    border: 1px solid var(--main-color)!important;
    border-radius: 12px!important;
    margin-bottom: 25px;
    padding: 23px 40px;
    box-shadow: 0px 3px 6px #4807E626;

}
.accordion .card-header{
    background-color: transparent;
    border: 0;
    border-radius: 8px;
    padding: 0;
    border: 1px solid var(--main-color)!important;
    border-radius: 12px!important;
    margin-bottom: 25px;
    padding: 23px 40px;
    box-shadow: 0px 3px 6px #4807E626;
}   
.card-body{
    padding: 0;
}
@media (max-width:1024px) {
    .faq__icon {
        display: none;
    }
}
@media (max-width:769px) {
   
}
@media (max-width:426px) {
    .faq{
        padding: 40px 0
    }
    .faq__heading{
        font-size: 30px;
    }
    .accordion .card{
        margin-bottom: 15px;
        padding: 15px 25px
    }
    
}
