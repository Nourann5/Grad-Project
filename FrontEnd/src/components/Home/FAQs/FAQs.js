import React from 'react'
import styles from './FAQs.module.css'
import './FAQs.css'
import {ReactComponent as FaqIconIcon} from 'assets/icons/faqIcon.svg'
import {ReactComponent as MinusIcon} from 'assets/icons/minus.svg'
import { Accordion, Container } from 'react-bootstrap'

function FAQs() {
    return (
    <div  className={`${styles["faq"]} faq`}>
        <Container>
            <div className={styles["faq__heading-cont"]}>
                <FaqIconIcon className={styles["faq__icon"]}/>
                <h2 className={styles["faq__heading"]}>FREQUENTLY ASKED QUESTIONS</h2>
            </div>
            <Accordion>
                <Accordion.Item eventKey="0" className={styles['faq__accordion-item']}>
                    <Accordion.Header className={styles['faq__accordion-header']}>
                            <div className={styles['faq__accordion-header-content']}>

                                Lorem ipsum dolor sit amet consetetur sadipscing ?
                                <span className={`${styles["faq__accordion-icon-cont"]}`}>
                                    <MinusIcon className={styles["faq__accordion-icon"]}/>
                                    <MinusIcon className={`${styles["faq__accordion-icon"]} faq__accordion-icon--open`}/>
                                </span>
                            </div>
                    <Accordion.Body className={styles['faq__accordion-body']}>
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
                    eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
                    minim veniam, quis nostrud exercitation ullamco laboris nisi ut
                    aliquip ex ea commodo consequat. Duis aute irure dolor in
                    reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
                    pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
                    culpa qui officia deserunt mollit anim id est laborum.
                    </Accordion.Body>
                    </Accordion.Header>
                </Accordion.Item>
                <Accordion.Item eventKey="1" className={styles['faq__accordion-item']}>
                    <Accordion.Header className={styles['faq__accordion-header']}>
                            <div className={styles['faq__accordion-header-content']}>

                                Lorem ipsum dolor sit amet consetetur sadipscing ?
                                <span className={`${styles["faq__accordion-icon-cont"]}`}>
                                    <MinusIcon className={styles["faq__accordion-icon"]}/>
                                    <MinusIcon className={`${styles["faq__accordion-icon"]} faq__accordion-icon--open`}/>
                                </span>
                            </div>
                    <Accordion.Body className={styles['faq__accordion-body']}>
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
                    eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
                    minim veniam, quis nostrud exercitation ullamco laboris nisi ut
                    aliquip ex ea commodo consequat. Duis aute irure dolor in
                    reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
                    pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
                    culpa qui officia deserunt mollit anim id est laborum.
                    </Accordion.Body>
                    </Accordion.Header>
                </Accordion.Item>

                <Accordion.Item eventKey="2" className={styles['faq__accordion-item']}>
                    <Accordion.Header className={styles['faq__accordion-header']}>
                            <div className={styles['faq__accordion-header-content']}>

                                Lorem ipsum dolor sit amet consetetur sadipscing ?
                                <span className={`${styles["faq__accordion-icon-cont"]}`}>
                                    <MinusIcon className={styles["faq__accordion-icon"]}/>
                                    <MinusIcon className={`${styles["faq__accordion-icon"]} faq__accordion-icon--open`}/>
                                </span>
                            </div>
                    <Accordion.Body className={styles['faq__accordion-body']}>
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
                    eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
                    minim veniam, quis nostrud exercitation ullamco laboris nisi ut
                    aliquip ex ea commodo consequat. Duis aute irure dolor in
                    reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
                    pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
                    culpa qui officia deserunt mollit anim id est laborum.
                    </Accordion.Body>
                    </Accordion.Header>
                </Accordion.Item>
            </Accordion>

        </Container>
        </div>

    
  )
}

export default FAQs