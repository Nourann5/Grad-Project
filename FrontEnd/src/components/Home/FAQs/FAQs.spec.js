import {render , screen} from '@testing-library/react'
import { BrowserRouter, Route } from 'react-router-dom'
import FAQs from './FAQs'

describe('Test the Home FAQS', () => { 
    it('Heading exists',()=>{
        const tree =  render(
            <BrowserRouter>
                <FAQs/>
            </BrowserRouter>
        )
        const headerTitle = screen.getByRole('heading',{name:'FREQUENTLY ASKED QUESTIONS'})
        expect(headerTitle).toBeInTheDocument()
        expect(tree).toMatchSnapshot()
    })
})