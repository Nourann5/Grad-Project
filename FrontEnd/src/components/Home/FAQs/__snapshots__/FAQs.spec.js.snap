// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Home FAQS Heading exists 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="faq faq"
      >
        <div
          class="container"
        >
          <div
            class="faq__heading-cont"
          >
            <svg
              class="faq__icon"
            >
              faqIcon.svg
            </svg>
            <h2
              class="faq__heading"
            >
              FREQUENTLY ASKED QUESTIONS
            </h2>
          </div>
          <div
            class="accordion"
          >
            <div
              class="faq__accordion-item accordion-item"
            >
              <h2
                class="faq__accordion-header accordion-header"
              >
                <button
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  type="button"
                >
                  <div
                    class="faq__accordion-header-content"
                  >
                    Lorem ipsum dolor sit amet consetetur sadipscing ?
                    <span
                      class="faq__accordion-icon-cont"
                    >
                      <svg
                        class="faq__accordion-icon"
                      >
                        minus.svg
                      </svg>
                      <svg
                        class="faq__accordion-icon faq__accordion-icon--open"
                      >
                        minus.svg
                      </svg>
                    </span>
                  </div>
                  <div
                    class="accordion-collapse collapse"
                  >
                    <div
                      class="faq__accordion-body accordion-body"
                    >
                      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                    </div>
                  </div>
                </button>
              </h2>
            </div>
            <div
              class="faq__accordion-item accordion-item"
            >
              <h2
                class="faq__accordion-header accordion-header"
              >
                <button
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  type="button"
                >
                  <div
                    class="faq__accordion-header-content"
                  >
                    Lorem ipsum dolor sit amet consetetur sadipscing ?
                    <span
                      class="faq__accordion-icon-cont"
                    >
                      <svg
                        class="faq__accordion-icon"
                      >
                        minus.svg
                      </svg>
                      <svg
                        class="faq__accordion-icon faq__accordion-icon--open"
                      >
                        minus.svg
                      </svg>
                    </span>
                  </div>
                  <div
                    class="accordion-collapse collapse"
                  >
                    <div
                      class="faq__accordion-body accordion-body"
                    >
                      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                    </div>
                  </div>
                </button>
              </h2>
            </div>
            <div
              class="faq__accordion-item accordion-item"
            >
              <h2
                class="faq__accordion-header accordion-header"
              >
                <button
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  type="button"
                >
                  <div
                    class="faq__accordion-header-content"
                  >
                    Lorem ipsum dolor sit amet consetetur sadipscing ?
                    <span
                      class="faq__accordion-icon-cont"
                    >
                      <svg
                        class="faq__accordion-icon"
                      >
                        minus.svg
                      </svg>
                      <svg
                        class="faq__accordion-icon faq__accordion-icon--open"
                      >
                        minus.svg
                      </svg>
                    </span>
                  </div>
                  <div
                    class="accordion-collapse collapse"
                  >
                    <div
                      class="faq__accordion-body accordion-body"
                    >
                      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                    </div>
                  </div>
                </button>
              </h2>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="faq faq"
    >
      <div
        class="container"
      >
        <div
          class="faq__heading-cont"
        >
          <svg
            class="faq__icon"
          >
            faqIcon.svg
          </svg>
          <h2
            class="faq__heading"
          >
            FREQUENTLY ASKED QUESTIONS
          </h2>
        </div>
        <div
          class="accordion"
        >
          <div
            class="faq__accordion-item accordion-item"
          >
            <h2
              class="faq__accordion-header accordion-header"
            >
              <button
                aria-expanded="false"
                class="accordion-button collapsed"
                type="button"
              >
                <div
                  class="faq__accordion-header-content"
                >
                  Lorem ipsum dolor sit amet consetetur sadipscing ?
                  <span
                    class="faq__accordion-icon-cont"
                  >
                    <svg
                      class="faq__accordion-icon"
                    >
                      minus.svg
                    </svg>
                    <svg
                      class="faq__accordion-icon faq__accordion-icon--open"
                    >
                      minus.svg
                    </svg>
                  </span>
                </div>
                <div
                  class="accordion-collapse collapse"
                >
                  <div
                    class="faq__accordion-body accordion-body"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </div>
                </div>
              </button>
            </h2>
          </div>
          <div
            class="faq__accordion-item accordion-item"
          >
            <h2
              class="faq__accordion-header accordion-header"
            >
              <button
                aria-expanded="false"
                class="accordion-button collapsed"
                type="button"
              >
                <div
                  class="faq__accordion-header-content"
                >
                  Lorem ipsum dolor sit amet consetetur sadipscing ?
                  <span
                    class="faq__accordion-icon-cont"
                  >
                    <svg
                      class="faq__accordion-icon"
                    >
                      minus.svg
                    </svg>
                    <svg
                      class="faq__accordion-icon faq__accordion-icon--open"
                    >
                      minus.svg
                    </svg>
                  </span>
                </div>
                <div
                  class="accordion-collapse collapse"
                >
                  <div
                    class="faq__accordion-body accordion-body"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </div>
                </div>
              </button>
            </h2>
          </div>
          <div
            class="faq__accordion-item accordion-item"
          >
            <h2
              class="faq__accordion-header accordion-header"
            >
              <button
                aria-expanded="false"
                class="accordion-button collapsed"
                type="button"
              >
                <div
                  class="faq__accordion-header-content"
                >
                  Lorem ipsum dolor sit amet consetetur sadipscing ?
                  <span
                    class="faq__accordion-icon-cont"
                  >
                    <svg
                      class="faq__accordion-icon"
                    >
                      minus.svg
                    </svg>
                    <svg
                      class="faq__accordion-icon faq__accordion-icon--open"
                    >
                      minus.svg
                    </svg>
                  </span>
                </div>
                <div
                  class="accordion-collapse collapse"
                >
                  <div
                    class="faq__accordion-body accordion-body"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </div>
                </div>
              </button>
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
